docs/reproducing-results.md

```diff
@@ -362,9 +362,88 @@ run inference through Nvidia NIM API.
        subprocess.run(cmd, shell=True, check=True)
    ```
 
-6. Now all the data is generated and you can follow up by converting it to the SFT format.
-
-   TBD
+6. Check for test set contamination.
+   We test against GSM8K, MATH, AMC 2023, and AIME 2024.  
+ 
+   Retrieve top-5 similar items from the test sets
+   ```
+   python -m nemo_skills.inference.retrieve_similar \
+      ++retrieve_from="./nemo_skills/dataset/gsm8k/test.jsonl ./nemo_skills/dataset/math/test.jsonl ./nemo_skills/dataset/amc23/test.jsonl ./nemo_skills/dataset/aime24/test.jsonl" \
+      ++compare_to="<path to workspace>/new-problems-solution-augmentation/**/output-rs0.jsonl" \
+      ++output_file=<path to workspace>/new-problems-solution-augmentation/contamination-retrieved.jsonl \
+      ++top_k=5
+   ```
+   > **_NOTE:_** Currently the above command doesn't run inside docker, so you will need to install additional packages.
+
+   Next, you need to run LLM inference to check those closest found problems from the output file. We use the Llama3.1-405B-Instruct model for this, and here's one way of doing it via Nvidia API catalog.
+
+    ```
+    ns check_contamination \
+        --cluster=local \
+        --input_file=<path to workspace>/new-problems-solution-augmentation/contamination-retrieved.jsonl \
+        --output_file=<path to workspace>/new-problems-solution-augmentation/contamination-llm.jsonl \
+        --server_type=openai \
+        --model=meta/llama-3.1-405b-instruct \
+        --server_address=https://integrate.api.nvidia.com/v1 \
+        ++check_both_ways=True;
+    ```
+    
+   Identify all the problems for which the `contaminated` key has the output True. 
+   Add the entry `"contaminated": True` in all the generation files in `<path to workspace>/new-problems-solution-augmentation/`. Here is a sample python script for this:
+
+    ```python
+    def load_contaminated_problems(jsonl_file):
+        contaminated_problems = set()
+        with open(jsonl_file, 'r') as f:
+            for line in f:
+                data = json.loads(line)
+                if data.get('contaminated', False):
```

Comment:
```suggestion
                if data['contaminated']:
```
Should always have this key as far as I understand. Better not have the default that might silently disable this filter if for some reason previous step didn't write these keys

Reply:
Good point!